float8_experimental/float8_dynamic_linear.py

```diff
@@ -55,27 +74,19 @@ def __init__(self, **super_kwargs):
         super().__init__(**super_kwargs)
 
     def forward(self, x):
-        # cast x to float8_e4m3fn if not using activation hooks
         x_fp8 = self.cast_to_float8_e4m3fn(x)
-
-        # cast w to float8_e4m3fn
-        w_fp8 = self.cast_to_float8_e4m3fn(self.weight)
-
+        if isinstance(self.weight, Float8Tensor):  # cast by FSDP
+            w_fp8 = self.weight
+        else:
+            w_fp8 = self.cast_to_float8_e4m3fn(self.weight)
         y = torch.nn.functional.linear(x_fp8, w_fp8, self.bias)
-
-        # Cast gradY to float8_e5m2 during backward if not using activation hooks
         y = self.cast_to_float8_e5m2_bw(y)
-
         return y
 
-    def cast_to_float8_e4m3fn(self, inpt_tensor: torch.Tensor) -> Float8Tensor:
-        if tensor_already_casted_to_fp8(inpt_tensor):
-            # check to early return if already casted to float8
-            return inpt_tensor
-        scale = tensor_to_scale(inpt_tensor, torch.float8_e4m3fn)
-        return Float8Tensor.to_float8(
-            inpt_tensor, scale, torch.float8_e4m3fn, mm_config=self.forward_config
-        )
+    def cast_to_float8_e4m3fn(
```

Comment:
it's unusual to have a regular function as well as a class method with the same name but different args, can we just have one?

Reply:
For consistency, I will also move `cast_to_float8_e5m2_bw` to be outside the `Float8DynamicLinear`. Otherwise, we have one cast function as standalone and the other as on the module.